### PR TITLE
shutdown ako on akoNS/avi-secret update or delete

### PR DESCRIPTION
**Pull request by shubhamavi**

### CODE REVIEW

**Comment by shubhamavi** [comment]
failed due to docker rate limit, will trigger the pipeline later

**Comment by shubhamavi** [comment]
build ako

**Review by monotosh-avi — COMMENTED** [review]
@shubhamavi  We don't want secret informers for open shift for now. Can we retain that ? 
Openshift has too many secrets by default, secret informers would take too much resources in open shift. 

Looks good to me otherwise

File: internal/k8s/controller.go [hunk 1]
```diff
@@ -475,54 +465,68 @@ func (c *AviController) SetupEventHandlers(k8sinfo K8sinformers) {
...
+					utils.AviLog.Errorf("Tombstone contained object that is not a Secret: %#v", obj)
 					return
 				}
 			}
-			namespace, _, _ := cache.SplitMetaNamespaceKey(utils.ObjKey(ingress))
-			key := utils.Ingress + "/" + utils.ObjKey(ingress)
-			bkt := utils.Bkt(namespace, numWorkers)
-			c.workqueue[bkt].AddRateLimited(key)
-			utils.AviLog.Debugf("key: %s, msg: DELETE", key)
+			if validateAviSecret(secret) {
```
Comment (monotosh-avi) [inline internal/k8s/controller.go:481]: Why if we delete an secret referred by an ingress, which is not in avi NS ?

**Review by monotosh-avi — COMMENTED** [review]
I think we need  to check if we can only sync secret from avi NS for openshift and from all Namespaces in non Openshift environment.

**Comment by shubhamavi** [comment]
@sudswasavi the code to initialise the informer for AKO specific ns is utils.go. checkout the akoNSInformerFactory code in utils.go

**Comment by sudswasavi** [comment]
> @sudswasavi the code to initialise the informer for AKO specific ns is utils.go. checkout the akoNSInformerFactory code in utils.go

Thanks. I missed seeing that. This will address @monotosh-avi 's concern as well I think.

**Comment by monotosh-avi** [comment]
@shubhamavi do you think this condition should be extended to open shift also, or is it fine ?
`
			if adv_l4 {
 				informers.SecretInformer = akoNSInformerFactory.Core().V1().Secrets()
 			} else {
 				informers.SecretInformer = kubeInformerFactory.Core().V1().Secrets()
 			}
`
@sudswasavi

**Comment by sudswasavi** [comment]
> @shubhamavi do you think this condition should be extended to open shift also, or is it fine ?
> `if adv_l4 { informers.SecretInformer = akoNSInformerFactory.Core().V1().Secrets() } else { informers.SecretInformer = kubeInformerFactory.Core().V1().Secrets() }`
> @sudswasavi

I think we should do it for openshift as well in this PR for completeness sake. Let's not init the secrets for openshift.

**Review by monotosh-avi — APPROVED** [review]

**Review by sudswasavi — APPROVED** [review]

**Review by sudswasavi — APPROVED** [review]

**Pull request merged**